### PR TITLE
Fix hdr align

### PR DESCRIPTION
**Pull request by kamleshbhalui**
Properly discard bits when casting to lower sizes
Fix typeMap for constants

### CODE REVIEW

File: backends/dpdk/dpdkArch.cpp [hunk 1]
```diff
@@ -474,6 +474,29 @@ bool CollectMetadataHeaderInfo::preorder(const IR::Type_Struct *s) {
...
+        bool all_hdr_field_aligned = true;
+        for (auto field : st->fields) {
+            if ((*field).type->is<IR::Type_Bits>()) {
+                auto t = (*field).type->to<IR::Type_Bits>();
+                auto width = t->width_bits();
+                size_sum_so_far += width;
+                if ((width & 0x7) != 0) {
+                    all_hdr_field_aligned = false;
+                }
+            }
```
Comment (mbudiu-vmw) [inline backends/dpdk/dpdkArch.cpp:488]: and if it's not bit<> you do nothing?

**Review by mbudiu-vmw — APPROVED** [review]

File: backends/dpdk/dpdkArch.cpp (hunk 1 above)
Comment (kamleshbhalui) [reply backends/dpdk/dpdkArch.cpp:488]: it's always bit<> at this point.

Comment (kamleshbhalui) [reply backends/dpdk/dpdkArch.cpp:488]: all fields will be of bit<> type at this point.

Comment (mbudiu-vmw) [reply backends/dpdk/dpdkArch.cpp:488]: that's what BUG is for.

Comment (kamleshbhalui) [reply backends/dpdk/dpdkArch.cpp:493]: yes done

**Pull request merged**